pinot-connectors/pinot-flink-connector/src/main/java/org/apache/pinot/connector/flink/sink/PinotSinkFunction.java

```diff
@@ -117,31 +123,22 @@ public void close()
   @Override
   public void invoke(T value, Context context)
       throws Exception {
-    _segmentWriter.collect(_recordConverter.convertToRow(value));
+    GenericRow row = _recordConverter.convertToRow(value);
+    _pendingRows.add(row);
+    _segmentWriter.collect(row);
     _segmentNumRecord++;
-    if (_segmentNumRecord > _segmentFlushMaxNumRecords) {
+    if (_segmentNumRecord >= _segmentFlushMaxNumRecords) {
```

Comment:
good catch :)

Reply:
That's what was happening in my integration test initially. The existing tests never hit the threshold, they were just depending on the automatic flush that happens in the close method

Reply:
:+1